spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -887,7 +918,14 @@ class DistriOptimizer[T: ClassTag] (
     // unpersist the model because the next time optimize is called, new `models` will be
     // created
     shutdown()
-    models.unpersist()
+
+    // reserve optimMethod internal state for each worker if need
+    if (reserveOptimMethod) {
+      previousOptim = models.map(m => m.optimMethods)
```

Comment:
Should we make a deep copy here?

Reply:
I think we should always unpersist models, as initThreadModels will always create new ones.